### DISCUSSION TITLE
New websocket ethernet switch built using Node.js

### DISCUSSION THREAD

**Discussion by krishenriksen** [copy/v86#777] [Show and tell]
So the [websockproxy](https://github.com/benjamincburns/websockproxy) is not the fastest in the world, we know this.

However I have just finished building a new one using Node.js.
This allows me to permessage-deflate compress the raw ethernet frames using zlibDeflate.

The current settings are close to default, as I haven't tested all variables for maximum throughput.

```JavaScript
const wss = new WebSocket.Server({
	port: 80,
	perMessageDeflate: {
		zlibDeflateOptions: {
			chunkSize: 1024,
			memLevel: 7,
			level: 3,
		},
		zlibInflateOptions: {
			chunkSize: 10 * 1024
		},
		clientNoContextTakeover: true, // Defaults to negotiated value.
		serverNoContextTakeover: true, // Defaults to negotiated value.
		serverMaxWindowBits: 10, // Defaults to negotiated value.
		// Below options specified as default values.
		concurrencyLimit: 10, // Limits zlib concurrency for perf.
		threshold: 1024, // Size (in bytes) below which messages should not be compressed.
	}
});
```

However if someone could test the speed compared against websockproxy and post here, it would be appreciated.

### node-relay
Use Docker, or build yourself for aarch64 architecture or any other using run.sh

[node-relay](https://github.com/krishenriksen/node-relay)

**Comment by dmgnr** [comment]
great lemme try

**Comment by Darin755** [comment]
I gave it a quick test and it seems that the old one is much faster for me
